lib/active_record/connection_adapters/oracle_enhanced_adapter.rb

```diff
@@ -569,6 +569,10 @@ def materialized_views #:nodoc:
         select_values("SELECT LOWER(mview_name) FROM all_mviews WHERE owner = SYS_CONTEXT('userenv', 'current_schema')")
       end
 
+      def synonym_names #:nodoc:
+        select_values("SELECT LOWER(synonym_name) FROM all_synonyms where owner = SYS_CONTEXT('userenv', 'current_schema')")
```

Comment:
Sounds good.  #1387 moved `synonyms` method from `SchemaStatementsExt` module to https://github.com/rsim/oracle-enhanced/blob/master/lib/active_record/connection_adapters/oracle_enhanced_adapter.rb#L573-L579 Since I overlooked `SchemaStatementsExt` module.

Would you remove `synonym_names` method here and change `data_sources` method as you suggested.